### PR TITLE
Update OHSketchTemplate.ino

### PR DESCRIPTION
**Pull request by Arribe**
## Description

Updated template with Doxygen clean-up improvements to assist with making clean documentation without misplaced comments.

Closes #

### Dependencies
* List any dependencies that are required for this change, including a full list of libraries required, especially if it is a new or otherwise unused library in the OpenHornet software.

### Type of change
- [ ] New software module (new software module for slave)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update outside of the automatically-generated Doxygen documentation.

### Checklist:
- [x] My code follows the [style guidelines](https://jrsteensen.github.io/OpenHornet-Software/d4/d46/md__2github_2workspace_2_s_t_y_l_e_g_u_i_d_e.html) of this project
- [x] [I have complied with the software manual](https://jrsteensen.github.io/OpenHornet-Software/d7/d78/md__software_manual.html) for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code fully with Doxygen compatible comments, particularly in hard-to-understand areas
- [x] I have made corresponding changes to non-Doxygen generated documentation
- [x] I have ran Doxygen locally, and it builds the docs successfully
- [x] My changes generate no errors on compile in Arduino IDE
- [x] My changes generate no new warnings on compile in Arduino IDE
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] (For sketches only) This sketch complies with OH-INTERCONNECT v**<insert version number here>**
- [ ] If this sketch requires additional libraries, [I have added it as a sub-module per the Arduino Libraries section of the Software Manual.](https://jrsteensen.github.io/OpenHornet-Software/d7/d78/md__software_manual.html)

### How Has This Been Tested?

- [ ] I have tested the sketch in-circuit in DCS with DCS-BIOS and outputs (displays, LEDs, etc.) function as expected. 
- [ ] I have tested the sketch in-circuit in DCS with DCS-BIOS and HID inputs (switches, pots, etc.) function as expected, with switches moving the correct direction.
- [ ] I have tested the sketch in-circuit in DCS with DCS-BIOS and any logic in the sketch has been tested and functions as expected.
- [x] This code has not yet been tested in-circuit.
- [ ] This code has not yet been tested in DCS-BIOS.

#### Description of Testing
N/A - base template compiles without warnings or errors.

#### Test Configuration
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

File: embedded/templates/OHSketchTemplate/OHSketchTemplate.ino [hunk 1]
```diff
@@ -45,25 +46,62 @@
...
+*/
 
 /**
- * Set DCS Bios to use irq serial
+ * Check if we're on a Mega328 or Mega2560 and define the correct
+ * serial interface
+ * 
  */
+#if defined(__AVR_ATmega328P__) || defined(__AVR_ATmega2560__)
 #define DCSBIOS_IRQ_SERIAL
```
Comment (jrsteensen) [inline embedded/templates/OHSketchTemplate/OHSketchTemplate.ino:71]:
```c
#define DCSBIOS_IRQ_SERIAL ///< This enables interrupt-driven serial communication for DCS-BIOS. (Only used with the ATmega328P or ATmega2560 microcontrollers.)
```

File: embedded/templates/OHSketchTemplate/OHSketchTemplate.ino [hunk 2]
```diff
@@ -45,25 +46,62 @@
...
+#else
+#define DCSBIOS_DEFAULT_SERIAL
```
Comment (jrsteensen) [inline embedded/templates/OHSketchTemplate/OHSketchTemplate.ino:73]:
```c
#define DCSBIOS_DEFAULT_SERIAL ///< This enables the default serial communication for DCS-BIOS. (Used with all other microcontrollers than the ATmega328P or ATmega2560.)  
```

File: embedded/templates/OHSketchTemplate/OHSketchTemplate.ino [hunk 3]
```diff
@@ -45,25 +46,62 @@
...
+#include <avr/power.h>
+#endif
 
 /**
- * DCS Bios library include
- */
+ * The Arduino pin that is connected to the
+ * RE and DE pins on the RS-485 transceiver.
+*/
+#define TXENABLE_PIN 5
```
Comment (jrsteensen) [inline embedded/templates/OHSketchTemplate/OHSketchTemplate.ino:84]:
```c
#define TXENABLE_PIN 5 ///< Sets TXENABLE_PIN to Arduino Pin 5
```

File: embedded/templates/OHSketchTemplate/OHSketchTemplate.ino [hunk 4]
```diff
@@ -45,25 +46,62 @@
...
+#define UART1_SELECT
```
Comment (jrsteensen) [inline embedded/templates/OHSketchTemplate/OHSketchTemplate.ino:85]:
```c
#define UART1_SELECT ///< Selects UART1 on Arduino for serial communication
```

**Review by jrsteensen — CHANGES_REQUESTED** [review]

**Comment by Arribe** [comment]
In backcheck.

**Review by jrsteensen — APPROVED** [review]

**Pull request merged**